lib/kickstart.rb

```diff
@@ -0,0 +1,254 @@
+      FileUtils.cp_r(content, output_dir, preserve: true)
+    end
+#    Executing profile method to write to ks.cfg file
+    profile(output_dir)
+    FileUtils.chmod(0600, File.join(output_dir, "ks.cfg"))
+    Machinery::Ui.puts(
+      "Note: The permssions of the KickStart directory are restricted to be" \
+        " only accessible by the current user. Further instructions are" \
+        " provided by the README.md in the exported directory."
+    )
+  end
+
+  def export_name
+    "#{@system_description.name}-kickstart"
+  end
+
+
+  def profile(output_dir)
+    File.open(File.join(output_dir, "ks.cfg"),"w") do |f|
+      f.write("lang en_us.UTF-8" +
+        "\ngraphical" +
+        "\ninstall" +
+        "\neula --agreed" +
+        "\nauth --enableshadow --passalgo=sha512" +
+        "\ncdrom" +
+        "\nfirstboot --enabled" +
+        "\nclearpart --all --initlabel" +
+        "\npart /boot --fstype ext4 --size=500" +
+        "\npart swap --size=1024" +
+        "\npart / --fstype ext4 --size=13480 --grow --asprimary" +
+        "\nkeyboard us" +
+        "\nrootpw --iscrypted $6$bNjXyviGZwK4X3yA$51rtDgkzad9CVE5KkRAltkQSiXoLJxY46oLLURaJ2mVcKEbDsf1tR3BmKQrk8hKS2fBerlBajBjGHXDBQRH261" +
+        "\nnetwork --bootproto=dhcp"+
+        "\nbootloader --append=\" crashkernel=auto\" --location=mbr" +
+        "\ntimezone Asia/Kolkata --isUtc")
+    end
+    File.open(File.join(output_dir, "ks.cfg"),"a") do |f|
+      f.write(apply_repositories)
+      f.write(apply_packages)
+      f.write(apply_users)
+      f.write(apply_groups)
+      f.write(apply_services_enabled)
+      f.write(apply_services_disabled)
+      apply_changed_files("changed_config_files")
+      apply_changed_files("changed_managed_files")
+      apply_unmanaged_files
+      f.write(apply_url_extraction)
+      f.write("\n%post --nochroot --log=/mnt/sysimage/root/post-install.log\n")
+      f.write(@chroot_scripts.join("\n"))
+      f.write("\n%end\n")
+    end
+ end
+
+  def outgoing_ip
+    output = Cheetah.run("ip", "route", "get", "8.8.8.8", stdout: :capture)
+    output[/ src ([\d\.:]+)\s*$/, 1] || "<ip>"
+  end
+
+  private
+
+  def check_exported_os
+    unless @system_description.os.is_a?(Machinery::OsRedhat)
+      raise Machinery::Errors::ExportFailed.new(
+        "Export is not possible because the operating system " \
+        "'#{@system_description.os.display_name}' is not supported."
+      )
+    end
+  end
+
+  def apply_repositories
+    return unless @system_description.repositories
+    ret_val=""
+        @system_description.repositories.each do |repository|
+        ret_val += "\nrepo --name=" + repository.alias.to_s + " --baseurl=" +  repository.url.to_s.chop.reverse.chop.reverse  
+      end
+    return ret_val
+    end
+
+  def apply_packages
+    return unless @system_description.packages
+    ret_val="\n%packages"
+      @system_description.packages.each do |package|
+        ret_val += "\n" + package.name.to_s
```

Comment:
For simple cases like this you could use `@system_description.packages.join("\n")`